floodlight/io/kinexon.py

```diff
@@ -0,0 +1,349 @@
+
+    links = {}
+    for gID in team_infos:
+        links.update(
+            {gID: {ID: xID for xID, ID in enumerate(team_infos[gID][identifier])}}
+        )
+
+    return links
+
+
+def read_kinexon_file(filepath_data: Union[str, Path], identifier: str = None) -> XY:
+    """Parse Kinexon files and extract position data.
+
+    Kinexon's local positioning system delivers one .csv file containing the position
+    data. This function provides a high-level access to Kinexon data by parsing "the
+    full file" given the path to the file.
+
+    Parameters
+    ----------
+    filepath_data: str of pathlib.Path
+        Full path to Kinexon.csv-file.
+    identifier: str
+        Column-name of personal identifier in Kinexon.csv-file.
+        Can be one of:
+            - 'sensor_id'
+            - 'mapped_id'
+            - 'name'
+
+    Returns
+    -------
+    positions: XY
+        XY-Object for the whole game. The order of groups is ascending according to
+        their group id. The order inside the groups is ascending according to their
+        appearance in the data.
+    """
+
+    # get metadata
+    team_infos, number_of_frames, frame_rate, t_null = get_meta_data(filepath_data)
+
+    # get links
+    links = create_links_from_meta_data(team_infos, identifier)
+    # get parameter-links
+    parameter_links = _get_parameter_links(filepath_data)
+
+    # create np.array for data
+    number_of_sensors = {}
+    xydata = {}
+    for group in links:
+        number_of_sensors.update({group: len(links[group])})
+        xydata.update(
+            {
+                group: np.full(
+                    [number_of_frames + 1, number_of_sensors[group] * 2], np.nan
+                )
+            }
+        )
+
+    # loop
+    with open(str(filepath_data), "r") as f:
+        while True:
+            line_string = f.readline()
+            # terminate if at end of file
+            if len(line_string) == 0:
+                break
+            line = line_string.split(",")
+            # skip the header of the file
+            if line[parameter_links["time"]].isdecimal():
+                timestamp = int(line[parameter_links["time"]])
+                # set group
+                group = line[parameter_links["group_id"]]
+                # set column
+                x_col = links[group][line[parameter_links["name"]]] * 2
+                y_col = x_col + 1
+                # set row
+                row = int((timestamp - t_null) / (1000 / frame_rate))
+
+                # set (x, y)-data
+                x_coordinate = line[parameter_links["x_coord"]]
+                y_coordinate = line[parameter_links["y_coord"]]
+
+                # insert (x, y)-data into column and row of respective array
+                if x_coordinate != "":
+                    xydata[group][row, x_col] = x_coordinate
+
+                if y_coordinate != "":
+                    xydata[group][row, y_col] = y_coordinate
+
+    # join group arrays
```

Comment:
I see now. We're changing this anyway soon, but for now maybe return a list of XY objects

Reply:
Returning list of XY objects for now